ocaml/xapi/xapi_udhcpd.ml

```diff
@@ -162,7 +162,18 @@ let restart_nolock () =
   let pid = try Unixext.pidfile_read !Xapi_globs.udhcpd_pidfile with _ -> None in
   Opt.iter Unixext.kill_and_wait pid;
   let (_: string * string) = execute_command_get_output !Xapi_globs.busybox [ "udhcpd"; !Xapi_globs.udhcpd_conf ] in
-  ()
+  let start = Mtime_clock.elapsed_ns () in
+  let rec wait_for_pid n =
+    let now = Mtime_clock.elapsed_ns () in
+    if Int64.sub now start > Int64.mul 30L 1000000000L then failwith "Failed to start udhcpd";
```

Comment:
I believe the Mtime module provides constants for this kind of conversion and also for measuring elapsed time. So this code could be made more idiomatic by using `Mtime.span` values. https://github.com/dbuenzli/mtime/blob/master/src/mtime_clock.mli